core/server/master/src/main/java/alluxio/master/block/meta/WorkerMetaLock.java

```diff
@@ -0,0 +1,67 @@
+package alluxio.master.block.meta;
+
+import com.google.common.collect.ImmutableList;
+
+import java.util.EnumSet;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.Condition;
+import java.util.concurrent.locks.Lock;
+
+public class WorkerMetaLock implements Lock {
+  private final EnumSet<WorkerMetaLockType> mLockTypes;
+  private final boolean mIsShared;
+  private final MasterWorkerInfo mWorker;
+  WorkerMetaLock(EnumSet<WorkerMetaLockType> lockTypes, boolean isShared, MasterWorkerInfo worker) {
+    mLockTypes = lockTypes;
+    mIsShared = isShared;
+    mWorker = worker;
+  }
+
+  @Override
+  public void lock() {
+    for (WorkerMetaLockType t : ImmutableList.of(WorkerMetaLockType.STATUS_LOCK,
+            WorkerMetaLockType.USAGE_LOCK, WorkerMetaLockType.BLOCKS_LOCK)) {
+      if (mLockTypes.contains(t)) {
+        if (mIsShared) {
+          mWorker.getLock(t).readLock().lock();
+        } else {
+          mWorker.getLock(t).writeLock().lock();
+        }
+      }
+    }
+  }
+
+  @Override
+  public void unlock() {
+    for (WorkerMetaLockType t : ImmutableList.of(WorkerMetaLockType.BLOCKS_LOCK,
+            WorkerMetaLockType.USAGE_LOCK, WorkerMetaLockType.STATUS_LOCK)) {
+      if (mLockTypes.contains(t)) {
+        if (mIsShared) {
+          mWorker.getLock(t).readLock().unlock();
+        } else {
+          mWorker.getLock(t).writeLock().unlock();
+        }
+      }
+    }
+  }
+
+  @Override
+  public void lockInterruptibly() throws InterruptedException {
+    throw new UnsupportedOperationException();
```

Comment:
I don't know how to properly implement these and don't think these are needed by the callers at this moment. So instead of inheriting `ReentrantLock` which implements these with internal logics, I'd rather they just throw when the callers are not calling in the way we support.

Reply:
Yea I think `LockResource` supporting multiple locks would be more elegant. Adding TODO there.